docs/apis/core/comment/index.md

```diff
@@ -0,0 +1,189 @@
+---
+title: Comment API
+tags:
+  - API
+---
+import { Since } from '@site/src/components';
+
+<Since versions={["2.0"]} />
+
+## Objectives
+
+The goals of Comment API:
+
+- Manage comments centrally
+- Use a consistent approach for all comments throughout Moodle
+- Easily integrate Comment API with existing modules
+- Works no matter JavaScript is enabled or not
+
+## Overview
+
+Comment API provides following functionalities:
+
+1. Add comments
+1. Manage comments
+1. Delete comments
+
+And provides a fancy ajax interface to add/delete comments without reloading page.
+
+## Comment API database table
+
+| Field | Type | Default | Info |
+| --- | --- | --- | --- |
+| id | int(10) | auto-incrementing | The unique ID for this comment. |
+| userid | int(10) |  | who wrote this comment |
+| contextid | int(10) |  | The context id defined in context table - identifies the instance of plugin owning the comment. |
+| itemid | int(10) |  | Some plugin specific item id (eg. forum post, blog entry or assignment submission) |
+| commentarea | int(10) |  | for example, in user profile, you can comment user's description or interests, but they share the same itemid(==userid), we need comment_area to separate them |
+| timecreated | int(10) |  |  |
+| timemodified | int(10) |  |  |
+| content | text |  | content of comment |
+
+## Use Comment API
+
+### Add an option to format_text function
+
+Using this `format_text` function will add a comment icon automatically at the end of the text:
+
+For example, using the following code in the forum module will add a comment icon to every post:
+
+```php
+$cmt = new stdclass;
+$cmt->contextid = $modcontext->id;
+$cmt->area      = 'format_post';
+$cmt->itemid    = $post->id;
+$options->comments = $cmt;
+echo format_text($post->message, $post->messageformat, $options, $course->id)."<hr />";
+```
```

Comment:
When migrating docs, we try to update to meet current coding style:

```suggestion
```php
$cmt = new stdClass();
$cmt->contextid = $modcontext->id;
$cmt->area = 'format_post';
$cmt->itemid = $post->id;
$options->comments = $cmt;
echo format_text($post->message, $post->messageformat, $options, $course->id) . "<hr />";
```
```